File: Programming Basics with JS/T06 - While Loop/exercise/03. Vacation.js
function vacation(input) {
    let moneyNeeded = Number(input[0]);
    let currentMoney = Number(input[1]);
    let text = '';
    let money = 0;
    let dayCount = 0;
    let spendDays = 0;
    let i = 2;
    let j = 3;

    while (currentMoney < moneyNeeded) {

        if (spendDays === 5) {
            console.log(`You can't save the money.`);
            console.log(`${dayCount}`);
            return;
        }
        text = input[i];
        money = Number(input[j]);

        if (text === 'spend') {
            currentMoney -= money;
            if (currentMoney < 0) {
                currentMoney = 0;
            }
            spendDays++;

        } else if (text === 'save') {
            currentMoney += money;
            spendDays = 0;
        }
        dayCount++;
        i += 2;
        j += 2;
    }
    console.log(`You saved the money for ${dayCount} days.`);
}
vacation(["2000", "1000", "spend", "1200", "save", "2000"]);
vacation(["110", "60", "spend", "10", "spend", "10", "spend", "10", "spend", "10", "spend", "10"]);
vacation(["250", "150", "spend", "50", "spend", "50", "save", "100", "save", "100"]);

File: Programming Basics with Go/T06 - Advanced Loops/exercise/03. Vacation.go
package main

import "fmt"

func main() {

	var tripPrice, currentMoney float32
	fmt.Scanln(&tripPrice)
	fmt.Scanln(&currentMoney)

	var countAllDays int = 0 //общо всички дни с действия
	var countSpendDays int = 0 //последователни дни, в които харчи

	//повтаряме: спестяване / харчене на пари
	//спираме: currentMoney >= tripPrice
	//продължавам: currentMoney < tripPrice

	for currentMoney < tripPrice {
		//спестявам или харча
		var action string
		fmt.Scanln(&action) //"spend" или "save"
		var sum float32
		fmt.Scanln(&sum) //пари за харчене / спестяване

		countAllDays++

		if action == "save" {
			//спестявам
			currentMoney += sum
			countSpendDays = 0
		} else if action == "spend" {
			//харчим
			countSpendDays++
			currentMoney -= sum

			//дали харчим повече отколкото имаме
			if currentMoney < 0 {
				currentMoney = 0
			}

			//дали 5 последователни дни на харчене
			if countSpendDays == 5 {
				fmt.Println("You can't save the money.")
				fmt.Println(countAllDays)
				break
			}
		}
	}

	if currentMoney >= tripPrice {
		//събира парите за екскурзията
		fmt.Printf("You saved the money for %d days.", countAllDays)
	}
}
